ui/scripts/vpc.js

```diff
@@ -745,12 +745,24 @@
                         }
                     },
                     dataProvider: function(args) {
+                        var array1 = [];
+                        if (args.filterBy != null) {
+                            if (args.filterBy.search != null && args.filterBy.search.by != null && args.filterBy.search.value != null) {
+                                if (args.filterBy.search.by === "name") {
+                                    if (args.filterBy.search.value.length > 0)
+                                        array1.push("&keyword=" + args.filterBy.search.value);
+                                }
+                            }
+                        }
+                        var data = {
+                            page: args.page,
+                            pageSize: pageSize,
+                            networkid: args.context.networks[0].id,
+                            listAll: true
+                        };
                         $.ajax({
-                            url: createURL('listLoadBalancers'),
-                            data: {
-                                networkid: args.context.networks[0].id,
-                                listAll: true
-                            },
+                            url: createURL('listLoadBalancers' + array1.join("")),
+                            data: data,
                             success: function(json) {
```

Comment:
@vedulasantosh Alright, I wanted you to explore better implementation. Here's one way, I would prefer (see how it remove the need of multiple-nested ifs):
```
                       var data = {
                            networkid: args.context.networks[0].id,
                            page: args.page,
                            pageSize: pageSize,
                            listAll: true
                       };
                       var keyword = (((args || {}).filterBy || {}).search || {}).value;
                       if (keyword) {
                           data.keyword = keyword;
                       }
                       # rest of the code
```